### ISSUE TITLE
latent Memory explosion code in run_epoch

### ISSUE THREAD

**Issue by xvshiting** [barronalex/Dynamic-Memory-Networks-in-TensorFlow#16]
` def run_epoch(self, session, data, num_epoch=0, train_writer=None, train_op=None, verbose=2, train=False):
        config = self.config
        dp = config.dropout
        if train_op is None:
            train_op = tf.no_op()
            dp = 1`

`loss, pred, summary, _ = session.run(
              [self.calculate_loss, self.pred, self.merged, train_op], feed_dict=feed)`
when train_op is None, the graph will add one no_op after running the run_epoch function. If we call this function a great times, the memory of machine will balloon.
